### DISCUSSION TITLE
Cache for graphql client - horizon app

### DISCUSSION THREAD

**Discussion by agsmartwolf** [swellstores/community#57] [Help] [answered]
Hi there,
I've started to use a horizon and i have a problem with cache.

I've updated products and fetched data via graphql client. The old response is returned. I don't have updated data.
But i have actual data in console in swell dashboard.

How can i figure it out?

**Comment by maarten2424** [comment]
Hey @agsmartwolf!
Could you explain how you deployed your horizon theme? Also; do you have these old responses client or server side?

**Reply by agsmartwolf** [reply]
It's about server side. I'm debugging locally. I'm using fork of horizon repo with code customisation of only frontend parts like components and pages layouts.

I've met also problems about footer menu items. I've added some links in swell editor a couple of days. And i'm still receiving old data with outdated links without new ones.

**Reply by maarten2424** [reply]
The GraphQL client does not cache the response from the Swell Editor locally, so i'm not sure what's going on there. How's the cache when deployed? Are you receiving updated links there?

**Reply by maarten2424** [reply]
Also @agsmartwolf could you provide us an example of the GraphQL Query?

**Comment by agsmartwolf** [comment, marked as answer]
Now it's working.
However now i can't find graphql schema info.

I need to extract menus depend on lang.
I tried the query:
```
query getMenus($locale: String) {
  menuSettings(_locale: $locale) {
    sections {
      id
      name
      items
    }
  }
}
```

But it's returned menu with `en` translation always.

**Reply by agsmartwolf** [reply]
Okay, I don't know why but the same query has worked!
```
query getMenus($locale: String) {
  menuSettings(_locale: $locale) {
    sections {
      id
      name
      items
    }
  }
}
```

So I mark topic as resolved. Thank you.
